src/manipulation.js

```diff
@@ -1,7 +1,21 @@
-var rxhtmlTag = /<(?!area|br|col|embed|hr|img|input|link|meta|param)(([a-z][^\/\0>\x20\t\r\n\f]*)[^>]*)\/>/gi;
+var rxhtmlTag = /<(?!area|br|col|embed|hr|img|input|link|meta|param)(([a-z][^\/\0>\x20\t\r\n\f]*)[^>]*)\/>/gi,
+	origHtmlPrefilter = jQuery.htmlPrefilter,
+	makeMarkup = function( html ) {
+		var doc = window.document.implementation.createHTMLDocument( "" );
+		doc.body.innerHTML = html;
+		return doc.body && doc.body.innerHTML;
+	};
 
 jQuery.UNSAFE_restoreLegacyHtmlPrefilter = function() {
 	jQuery.htmlPrefilter = function( html ) {
 		return html.replace( rxhtmlTag, "<$1></$2>" );
 	};
 };
+
+jQuery.htmlPrefilter = function( html ) {
```

Comment:
Restoring the legacy filter above will clobber this warning, but it's already marked `UNSAFE_` so we can do just about anything we want including format the drive.

Reply:
That's true. That said, the part below with the `makeMarkup` comparison & the printing the warning could be extracted to a separate function which we could then use both here & in the `htmlPrefilter` version that `UNSAFE_restoreLegacyHtmlPrefilter` restores. What do you think?

Reply:
I wonder if someone who uses the UNSAFE_ call actually wants the warning? Probably not.

Reply:
@dmethvin I think many of them do! Some people may not care (but if they don't care about Migrate warnings they may just mute them all) but I imagine many users of this API won't be people _not_ wanting to adjust but ones that want to perform the jQuery update but who don't have time (or possibilities - they may rely on external code that needs to get updated) to deal with that breaking change at the moment. They will often still want to fix their jQuery usage & remove the `UNSAFE_` call. Migrate warnings would help them a lot here!

Reply:
I think people that want this warning silenced without silencing the others & who want the old behavior restored can follow the non-Migrate based advice from our [3.5 upgrade guide](https://jquery.com/upgrade-guide/3.5/).

But I'd expect many users of this `UNSAFE_` API to want to be warned so that they can fix violations & get rid of it as it essentially undoes a security fix.

Reply:
You're right! I could see someone restoring the filter temporarily and letting things run to see what warnings they got. Without that, their code would be broken and they might not be able to test things fully to fix them. I've added a commit, see what you think.